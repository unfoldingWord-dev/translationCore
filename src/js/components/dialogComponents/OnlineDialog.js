import React from 'react';
import PropTypes from 'prop-types';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Checkbox } from 'material-ui';
import { withLocale } from '../../containers/Locale';
import { connect } from 'react-redux';

const OnlineDialog = ({translate, checked, onChecked}) => {
  return (
    <MuiThemeProvider>
      <div>
        <p style={{fontSize: 15}}>
          {translate('network_warning')}
        </p>
        <div style={{display: 'flex'}}>
          <Checkbox
            style={{width: '0px', marginRight: -10}}
            iconStyle={{fill: 'black'}}
            checked={checked}
            labelStyle={{
              color: 'var(--reverse-color)',
              opacity: '0.7',
              fontWeight: '500'
            }}
            onCheck={(e) => {
              onChecked(e.target.checked);
            }}
          />
          {translate('hide_warning')}
        </div>
      </div>
    </MuiThemeProvider>
  );
};
OnlineDialog.propTypes = {
  translate: PropTypes.func,
  checked: PropTypes.bool,
  onChecked: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
  checked: state.settingsReducer.onlineMode
});

export default withLocale(connect(mapStateToProps)(OnlineDialog));
