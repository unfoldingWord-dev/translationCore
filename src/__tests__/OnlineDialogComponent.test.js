/* eslint-env jest */
import React from 'react';
// Mock store set up
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import renderer from 'react-test-renderer';
import OnlineDialog from '../js/components/dialogComponents/OnlineDialog';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
jest.mock('material-ui/Checkbox');

// Tests for OnlineDialog React Component
describe('OnlineDialog Component', () => {
  test('Should render a component that matches snapshot', () => {
    const props = {
      translate: key => key,
      checked: 'false',
      onChecked: jest.fn(),
    };
    const state = { settingsReducer: { onlineMode: false } };
    const store = mockStore(state);

    const tree = renderer.create(
      <OnlineDialog store={store} {...props} />
    );
    expect(tree).toMatchSnapshot();
  });
});
