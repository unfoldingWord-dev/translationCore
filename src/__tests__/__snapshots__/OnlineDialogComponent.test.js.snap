// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnlineDialog Component Should render a component that matches snapshot 1`] = `
<div>
  <p
    style={
      Object {
        "fontSize": 15,
      }
    }
  >
    using_internet
  </p>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    do_not_show_again
  </div>
</div>
`;
